src/gateway/Gateway.sol

```diff
@@ -74,22 +73,14 @@ contract Gateway is Auth, IGateway {
     }
 
     // --- Outgoing ---
-    function send(bytes calldata message) public pauseable {
+    function send(bytes calldata message, bool isPrepaid) public pauseable {
         require(
             msg.sender == investmentManager || msg.sender == poolManager || msg.sender == messages[message.toUint8(0)],
             "Gateway/invalid-manager"
         );
 
-        if (MessagesLib.messageType(message) == MessagesLib.Call.EntryPoint) {
-            entryPoint = message.toAddress(1);
-            return;
-        }
-
-        aggregator.send{value: address(this).balance}(message);
-
-        if (entryPoint != address(0)) {
-            delete entryPoint;
-        }
+        uint256 gasFunds = isPrepaid ? address(this).balance : 0;
```

Comment:
I don't think that this line is correct as in both cases you would use the balance, but for prepaid tx we somehow need to make sure that the user sent enough funds.

So sth like: 
`uint256 gasFunds = isPrepaid ? userGasBalance : address(this).balance;`